ld/emultempl/riscvelf.em

```diff
@@ -39,7 +43,16 @@ riscv_elf_before_allocation (void)
   else
     ENABLE_RELAXATION;
 
-  link_info.relax_pass = 2;
+  /* check if all of input files are binary blob.
+     If true, I skip relaxation pass. */
+  for (abfd = link_info.input_bfds; abfd != NULL; abfd = abfd->link.next) {
+    if (abfd->xvec != &binary_vec) {
+      all_inputs_binary = FALSE;
+      break;
+    }
+  }
+
+  link_info.relax_pass = all_inputs_binary == TRUE? 0: 2;
```

Comment:
This seems like a hack.  Shouldn't this be done in _bfd_riscv_relax_section, skipping each individual pass if binary?  Comparing against &binary_vec also seems brittle.

In relax_section, there's a check that bails out if reloc_count == 0.  Why isn't that triggered for binary sections?